bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3671,21 +3671,21 @@ void ReplicaImp::onMessage<ReqMissingDataMsg>(ReqMissingDataMsg *msg) {
       }
     }
 
-    if (msg->getFullCommitIsMissing()) {
+    if (msg->getFullCommitIsMissing() && seqNumInfo.isTimeCorrect()) {
       CommitFullMsg *c = seqNumInfo.getValidCommitFullMsg();
       if (c != nullptr) {
         LOG_INFO(CNSUS, "Sending FullCommit message as a response of RFMD" << KVLOG(msgSender, msgSeqNum));
         sendAndIncrementMetric(c, msgSender, metric_sent_commitFull_msg_due_to_reqMissingData_);
       }
     }
 
-    if (msg->getFullCommitProofIsMissing() && seqNumInfo.hasFastPathFullCommitProof()) {
+    if (msg->getFullCommitProofIsMissing() && seqNumInfo.hasFastPathFullCommitProof() && seqNumInfo.isTimeCorrect()) {
       FullCommitProofMsg *fcp = seqNumInfo.getFastPathFullCommitProofMsg();
       LOG_INFO(CNSUS, "Sending FullCommitProof message as a response of RFMD" << KVLOG(msgSender, msgSeqNum));
       sendAndIncrementMetric(fcp, msgSender, metric_sent_fullCommitProof_msg_due_to_reqMissingData_);
     }
 
-    if (msg->getPartialProofIsMissing()) {
+    if (msg->getPartialProofIsMissing() && seqNumInfo.isTimeCorrect()) {
       // TODO(GG): consider not to send if msgSender is not a collector
       PartialCommitProofMsg *pcf = seqNumInfo.getFastPathSelfPartialCommitProofMsg();
```

Comment:
should we expect partial commit proof msg as nullptr if time is incorrect ?

Reply:
Since we are creating next messages in consensus but not sending them only, we are not getting partialproof as nullptr.